### ISSUE TITLE
Refactor: Generalize Forms and Functions

### ISSUE THREAD

**Issue by ash1eygrace** [ash1eygrace/ai-content#7]
The goal is to generalize and abstract the form and reusable function and create each component easily without having to copy/paste. 

### Possible new structure:

```
components
├── miniapps*
│   ├── BlogIntro.js
│   ├── CompanyBio.js
│   ├── Home.js
│   ├── LinkedInJobDescription.js
│   └── ProductDescription.js
├── general
│   ├── Navigation.js
│   ├── Form.js
│   └── CallAPI.js
```

___ 

### Considerations: 

- there will **always be at least one form input field** 
- there can be >1 amount(s) of forms (determined by the individual components )
- fix your frakking nesting it's horrible. 
- the form submission data needs to be stored in a variable that will pass though the reusable function: 
	-  `Form onSubmit={this.onFormSubmit}` calls `onFormSubmit = e =>`  but we need to change that function to store the place holder text into a new var that passes through the promp. e.g. `prompt: "Write a persuasive and exciting product description for: ${formDataObj.productName}"`,** needs to be `prompt: newVarPrompt` 
___ 

### Sidenotes: 
- Might make sense to keep the form and resuable function within the same file just abstract the mini apps into their own components to pass information through the form, at least for now,  given that we're just using general API calls. Once we build a UI for the completion model it will make sense to abstact and the API call further into a reuable function t. We'll see. 
- Each component is still `class ProductDescription extends Component {` and pulled into navigation.js via router.  

___

### Form examples: 

Currently one has two inputs and the others have one input. 

**####One input:** 
``` JavaScript
<Form onSubmit={this.onFormSubmit}>
	<Form.Group className="mb-3" controlId="textArea">
	<Form.Label>Job Title:</Form.Label>
		<Form.Control as="textarea" name="jobTitle" placeholder="e.g.Website Technical Support Specialist" />
	</Form.Group>
<Button variant="primary" size="md" type="submit">Submit</Button>
</Form>
```

**####Two inputs:**
``` javascript
<Form onSubmit={this.onFormSubmit}>

<Form.Group className="mb-3" controlId="textArea">
	<Form.Label>Blog post title: </Form.Label>
	<Form.Control as="textarea" name="blogTitle" placeholder="e.g. How to Make a Commit with Git" />
</Form.Group>

<Form.Group className="mb-3" controlId="textArea">
	<Form.Label>SEO keywords: </Form.Label>
	<Form.Control as="textarea" name="context" placeholder="e.g. Git, CLI, commit message" />
</Form.Group>

<Button variant="primary" size="md" type="submit">
Submit
</Button>

</Form>
```

**Prompt:**
``` JavaScript

prompt: `Write an uplifting and positive Blog intro paragraph for the blog title ${formDataObj.blogTitle} and include the keywords: ${formDataObj.context}`,
```

### Form variable examples: 

<Form.Label>**Product Name & Purpose:**</Form.Label> 

name="**productName**" 

placeholder="**e.g. ScoobySnacks will make your dog chill out**"

### Related:

#3 #4

**Comment by ash1eygrace** [comment]
For generalizing forms using formik and yup for validation:

- formik: https://www.npmjs.com/package/formik 
- example: https://www.pluralsight.com/guides/generating-dynamic-forms-from-json-in-react 
- example: https://www.velotio.com/engineering-blog/using-formik-to-build-dynamic-forms-in-react

**Comment by ash1eygrace** [comment]
This was solved in Refactor api function #15 

Decided against the form abstraction and went with the async exported function.
